### PR TITLE
Add css grid layout to the book details

### PR DESCRIPTION
**Pull request by JuanGutierrez43**
## Summary 

This will add the redesign of the web page with the use of the CSS grid. For this feature I used grid areas to set the divs in the same way that the invision model. 

(I recreated this card because when I merged with master my divs just died. So I needed to delete the branch and start this feature from 0.)

**New features**

- Add grid for the book details
- Removed flex use where I implemented the grid. 
- Add new desing for book details with grid. 
- Add new mobile design for book details using grid too.
- Fixed some divs with problems on the use of the grid. 
- Fixed some margin problems. 

## Screenshots 
Screenshot from 2020-04-16 13-55-34
Screenshot from 2020-04-16 13-55-46

## Cards

https://trello.com/c/ftRTatey/12-redise%C3%B1o-de-la-p%C3%A1gina

### CODE REVIEW

File: scss/detail.scss [hunk 1]
```diff
@@ -38,18 +38,24 @@ $lower-screen-width: calc(#{$book-badge-width} * 0.7);
 .book-cover-page {
   background-color: $white;
   box-shadow: 4px 4px 10px 0 $black-shadow;
-  display: flex;
-  height: 416px;
+  display: grid;
+  grid-template-areas: 
+    "area1 area1"
+    "area2 area2"
+    "area3 area4";
```
Comment (eledwinn) [inline scss/detail.scss]: Please add semantic names to areas, like `book-image`, `title`, etc. try use same or similar class name.

File: scss/detail.scss [hunk 2]
```diff
@@ -73,33 +79,52 @@ $lower-screen-width: calc(#{$book-badge-width} * 0.7);
...
+}
+
+.book-genre-container {
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  grid-area: area2;
+}
+
+.book-img-container{
```
Comment (eledwinn) [inline scss/detail.scss]:
```suggestion
.book-img-container {
```

**Review by eledwinn — CHANGES_REQUESTED** [review]

File: scss/detail.scss [hunk 3]
```diff
@@ -38,18 +38,24 @@ $lower-screen-width: calc(#{$book-badge-width} * 0.7);
...
+  grid-template-areas: 
+    "area1 area1"
+    "area2 area2"
+    "area3 area4";
+  height: 569px;
+  margin-left: 35px;
   position: relative;
-  margin: 50px 0 0 50px;
-  width: 900px;
+  width: 853px;
```
Comment (CrisVega08) [inline scss/detail.scss]:
```suggestion
  width: 855px.  o. width: 850px. ;
```

File: scss/detail.scss [hunk 4]
```diff
@@ -38,18 +38,24 @@ $lower-screen-width: calc(#{$book-badge-width} * 0.7);
...
   background-color: $white;
   box-shadow: 4px 4px 10px 0 $black-shadow;
-  display: flex;
-  height: 416px;
+  display: grid;
+  grid-template-areas: 
+    "area1 area1"
+    "area2 area2"
+    "area3 area4";
+  height: 569px;
```
Comment (CrisVega08) [inline scss/detail.scss]:
```suggestion
  height: 570px;
```

File: scss/detail.scss [hunk 5]
```diff
@@ -73,33 +79,52 @@ $lower-screen-width: calc(#{$book-badge-width} * 0.7);
...
   font-weight: 700;
   line-height: 30px;
+  margin-bottom: 50px;
+}
+
+.book-genre-container {
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  grid-area: area2;
```
Comment (CrisVega08) [inline scss/detail.scss]: Same comment of edwin

File: scss/detail.scss [hunk 6]
```diff
@@ -73,33 +79,52 @@ $lower-screen-width: calc(#{$book-badge-width} * 0.7);
...
+}
+
+.book-img-container{
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  grid-area: area3;
```
Comment (CrisVega08) [inline scss/detail.scss]: Same

File: scss/detail.scss [hunk 7]
```diff
@@ -73,33 +79,52 @@ $lower-screen-width: calc(#{$book-badge-width} * 0.7);
...
 }
 
 .book-header {
-  border-bottom: 5px solid $earls-green;
-  margin: 10px 0 15px 15px;
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  grid-area: area1;
```
Comment (CrisVega08) [inline scss/detail.scss]: Same

File: scss/detail.scss [hunk 8]
```diff
@@ -73,33 +79,52 @@ $lower-screen-width: calc(#{$book-badge-width} * 0.7);
...
 }
 
 .book-info {
+  align-self: center;
   display: flex;
+  flex: 1;
   flex-direction: column;
-  flex-grow: 1;
-  margin: 0 25px 0 10px;
+  grid-area: area4;
```
Comment (CrisVega08) [inline scss/detail.scss]: Same

File: scss/detail.scss [hunk 9]
```diff
@@ -266,19 +291,26 @@ $lower-screen-width: calc(#{$book-badge-width} * 0.7);
...
   .book-detail-container {
     align-items: center;
+    display: flex;
     flex-direction: column-reverse;
   }
 
   .book-cover-page {
-    flex-direction: column;
+    grid-template-areas: 
+      "area1 area1"
```
Comment (CrisVega08) [inline scss/detail.scss]: same

File: scss/detail.scss [hunk 10]
```diff
@@ -266,19 +291,26 @@ $lower-screen-width: calc(#{$book-badge-width} * 0.7);
...
     height: 570px;
     margin: 50px 0 0;
     padding: 20px;
     width: 335px;
-
+  
     &::after {
       height: $lower-screen-height;
       width: $lower-screen-width;
+      top: 60px;
```
Comment (CrisVega08) [inline scss/detail.scss]: sort properties

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

**Review by eledwinn — APPROVED** [review]

**Review by CrisVega08 — APPROVED** [review]

**Pull request merged**